packages/console/src/pages/SignInExperience/index.tsx

```diff
@@ -40,18 +45,31 @@ const SignInExperience = () => {
     }
   }, [data, reset]);
 
-  const onSubmit = handleSubmit(async (formData) => {
-    if (!data || isSubmitting) {
-      return;
-    }
-
+  const saveData = async () => {
     const updatedData = await api
       .patch('/api/sign-in-exp', {
-        json: signInExperienceParser.toRemoteModel(formData),
+        json: signInExperienceParser.toRemoteModel(getValues()),
       })
       .json<SignInExperienceType>();
     void mutate(updatedData);
     toast.success(t('application_details.save_success'));
+  };
+
+  const onSubmit = handleSubmit(async (formData) => {
+    if (!data || isSubmitting) {
+      return;
+    }
+
+    const formatted = signInExperienceParser.toRemoteModel(formData);
+
+    // Sign in methods changed, need to show confirm modal first.
+    if (!compareSignInMethods(data, formatted)) {
+      setDataToCompare(formatted);
+
+      return;
+    }
+
+    await saveData();
```

Comment:
Do we have to await?

Reply:
No, we don't have to, but since is inside an "async function", so just await to keep the same behavior.